digits/model/tasks/caffe_train.py

```diff
@@ -444,23 +401,22 @@ def save_files_classification(self):
             shape.dim.append(self.dataset.image_dims[0])
             shape.dim.append(self.dataset.image_dims[1])
 
-        # hidden layers
-        deploy_network.MergeFrom(hidden_layers)
-
-        # output layers
-        if loss_layers[-1].type == 'SoftmaxWithLoss':
-            prob_layer = deploy_network.layer.add(
-                    type = 'Softmax',
-                    name = 'prob')
-            prob_layer.bottom.append(network_outputs[-1])
-            prob_layer.top.append('prob')
+        # Layers
+        deploy_network.MergeFrom(deploy_layers)
 
+        # Write to file
         with open(self.path(self.deploy_file), 'w') as outfile:
             text_format.PrintMessage(deploy_network, outfile)
 
         # network sanity checks
         self.logger.debug("Network sanity check - deploy")
         CaffeTrainTask.net_sanity_check(deploy_network, caffe_pb2.TEST)
+        found_softmax = False
+        for layer in deploy_network.layer:
+            if layer.type == 'Softmax':
+                found_softmax = True
+                break
+        assert found_softmax, 'Your deploy network is missing a Softmax layer! Read the documentation for custom networks and/or look at the standard networks for examples.'
```

Comment:
Did I miss the bit in the documentation where it is explained to the user that a `softmax` layer is needed to display a probability distribution?